### PR TITLE
fix(nuxt): handle errors loading app manifest

### LINKED ISSUE #24229: Error when opening nuxt3 web link in China's QQ application
### Environment

- Operating System: Darwin
- Node Version:     v20.9.0
- Nuxt Version:     3.8.1
- CLI Version:      3.9.1
- Nitro Version:    2.7.2
- Package Manager:  pnpm@8.10.0
- Builder:          -
- User Config:      alias, app, spaLoadingTemplate, runtimeConfig, plugins, modules, vant, css, postcss, nitro, vite, vue, telemetry, sourcemap
- Runtime Modules:  @vant/nuxt@1.0.3, @pinia-plugin-persistedstate/nuxt@1.2.0, @pinia/nuxt@0.5.1
- Build Modules:    -

### Reproduction

The link I used is https://m.xchuxing.com

### Describe the bug

At present, it is only found that an error will be reported when nuxt3 is opened in QQ's webview. The error message is as shown in the figure.The link I used is https://m.xchuxing.com
IMG_0277
Then I located ju.matchAll by controlling the search and found the source code in the map file.
image
I didn't use pages. I deleted it, created a view, and customized the route through app-router.options.ts. I wonder if it has something to do with this operation, but it works in other browsers, only QQ. The webview will report an error, how to solve it?

### Additional context

_No response_

### Logs

_No response_

### LINKED ISSUE #24625: nuxt3.8.X + tauri : "Cannot read properties of undefined (reading 'entries')"
### Environment

nuxi info :
```
- Operating System: `Darwin`
- Node Version:     `v18.18.2`
- Nuxt Version:     `3.8.2`
- CLI Version:      `3.10.0`
- Nitro Version:    `2.8.1`
- Package Manager:  `pnpm@8.10.5`
- Builder:          `-`
- User Config:      `devtools`, `ssr`, `devServer`
- Runtime Modules:  `-`
- Build Modules:    `-`
```

tauri info:
```
✔] Environment
    - OS: Mac OS 14.1.2 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.74.0 (79e9716c9 2023-11-13)
    ✔ cargo: 1.74.0 (ecb9851af 2023-10-18)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-apple-darwin (default)
    - node: 18.18.2
    - pnpm: 8.10.5
    - yarn: 1.22.19
    - npm: 9.8.1
    - bun: 1.0.0

[-] Packages
    - tauri [RUST]: 2.0.0-alpha.18
    - tauri-build [RUST]: 2.0.0-alpha.12
    - wry [RUST]: 0.34.2
    - tao [RUST]: 0.23.0
    - tauri-cli [RUST]: 2.0.0-alpha.9
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 2.0.0-alpha.18

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: 
    - devPath: http://localhost:1420/
    - framework: Vue.js (Nuxt)
    - bundler: Webpack

[-] iOS
    - Developer Teams: retracted
```

### Reproduction

https://github.com/flapili/repro-nuxt3.8-tauri

### Describe the bug

image

### Additional context

I thinks it's due to how tauri handle http ( proxied via `tauri.localhost/`) ? also if I downgrade nuxt to 3.7.4 it work

### Logs

```shell-script
nuxt.js?v=d306a4ef:97 [nuxt] error caught during app initialization TypeError: Cannot read properties of undefined (reading 'entries')
    at _sortRoutesMap (index.mjs?v=d306a4ef:218:16)
    at _matchRoutes (index.mjs?v=d306a4ef:200:30)
    at Object.matchAll (index.mjs?v=d306a4ef:158:25)
    at getRouteRules (manifest.js?v=d306a4ef:27:30)
    at async manifest-route-rule.js?v=d306a4ef:8:89
    at async router.js?v=d306a4ef:126:26
```

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/24625
resolves https://github.com/nuxt/nuxt/issues/24229

### 📚 Description

This adds some error handling and fallback if there is an issue fetching the latest app manifest.

The (unreleased) move from appConfig -> runtimeConfig should also make this stabler by moving `buildId` -> `runtimeConfig` in any case.

### CODE REVIEW

File: packages/nuxt/src/app/composables/manifest.ts [hunk 1]
```diff
@@ -24,9 +24,14 @@ function fetchManifest () {
...
   }
-  manifest = $fetch<NuxtAppManifest>(buildAssetsURL(`builds/meta/${useRuntimeConfig().app.buildId}.json`))
+  manifest = $fetch<NuxtAppManifest>(buildAssetsURL(`builds/meta/${useRuntimeConfig().app.buildId}.json`), {
+    responseType: 'json'
+  })
   manifest.then((m) => {
     matcher = createMatcherFromExport(m.matcher)
+  }).catch((err) => {
+    console.error('[nuxt] Error while fetching app manifest:', err)
+    throw err
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/manifest.ts:34]:
```suggestion
```

**Review by pi0 — APPROVED** [review]
Nice! I faced it with few investigations in past...

**Review by TheAlexLichter — APPROVED** [review]

**Comment by jim55167** [comment]
Node.js version:  v2.16.0
Nuxt.js version:  3.13.0
Hosting: IIS

I'm still encountering 'Error fetching app manifest. FetchError: [GET] "/_nuxt/builds/meta/xxxxxx.json"' in the production environment.

螢幕擷取畫面 2024-08-27 113924

**Comment by jim55167** [comment]
After adding a simple page, opening the page in the production environment shows a 404 error.
螢幕擷取畫面 2024-08-27 135402
螢幕擷取畫面 2024-08-27 135724

**Comment by MechaGirls** [comment]
My version "nuxt": "3.17.7",

`
Failed to load resource: the server responded with a status of 404 ()
BnGI3YRB.js:23 [nuxt] Error fetching app manifest. FetchError: [GET] "/_nuxt/builds/meta/3c967d01-d5bc-4e96-bcbb-5349226800cb.json": 404 
    at async s (BnGI3YRB.js:19:23854)
(anonymous) @ BnGI3YRB.js:23
BnGI3YRB.js:19 [nuxt] error caught during app initialization vd: [GET] "/_nuxt/builds/meta/3c967d01-d5bc-4e96-bcbb-5349226800cb.json": 404 Caused by: FetchError: [GET] "/_nuxt/builds/meta/3c967d01-d5bc-4e96-bcbb-5349226800cb.json": 404 
    at async s (BnGI3YRB.js:19:23854)
(anonymous) @ BnGI3YRB.js:19
BnGI3YRB.js:14 Hydration completed but contains mismatches.
`

**Pull request merged**